src/index.ts

```diff
@@ -28,62 +51,138 @@ export const defaultCompileOptions = (outputOptions: OutputOptions): CompileOpti
+      new StrictTransform(context, entry, outputOptions),
+    ];
   }
 
-  return options;
+  return null;
 };
 
-export default function closureCompiler(compileOptions: CompileOptions = {}): Plugin {
+/**
+ * Invoke all transformations for this Plugin invocation.
+ * @param transforms Bound Transformations instances to apply
+ * @param code source code to transform
+ * @param id Rollup's id entry for this source
+ * @return Transformed source from all transforms associated with this Plugin invocation.
+ */
+export const transform = async (
+  transforms: Array<Transform> | null,
+  code: string,
+  id: string,
+): Promise<TransformSourceDescription> => {
+  let output: TransformSourceDescription = {
+    code,
+  };
+
+  if (transforms) {
+    for (let iterator = 0; iterator < transforms.length; iterator++) {
+      const transformed = await transforms[iterator].input(output.code, id);
+      if (transformed) {
+        output = transformed;
+      }
+    }
+  }
+
+  return output;
+};
+
+/**
+ * Transform the tree-shaken code from Rollup with Closure Compiler (with derived configuration and transforms)
+ * @param compileOptions Closure Compiler compilation options from Rollup configuration.
+ * @param transforms Transforms to apply to source followin Closure Compiler completion.
+ * @param code Source to compile.
+ * @param outputOptions Rollup Output Options.
+ * @return Closure Compiled form of the Rollup Chunk
+ */
+export const transformChunk = async (
+  compileOptions: CompileOptions,
+  transforms: Array<Transform> | null,
+  code: string,
+  outputOptions: OutputOptions,
+): Promise<{ code: string; map: RawSourceMap } | void> => {
+  const jsAsFile = sync(code);
+  const mapAsFile = sync('');
+
+  compileOptions = Object.assign(defaultCompileOptions(transforms, outputOptions), compileOptions, {
+    js: jsAsFile,
+    create_source_map: mapAsFile,
+  });
+
+  const compile: Promise<string> = new Promise((resolve: (stdOut: string) => void, reject: (error: any) => void) => {
+    new compiler(compileOptions).run(async (exitCode: number, stdOut: string, stdErr: string) => {
+      if (exitCode !== 0) {
+        reject(new Error(`Google Closure Compiler exit ${exitCode}: ${stdErr}`));
+      } else {
+        if (transforms) {
+          for (let iterator = 0; iterator < transforms.length; iterator++) {
+            const transformed = await transforms[iterator].output(stdOut, 'none');
+            if (transformed && transformed.code) {
+              stdOut = transformed.code;
+            }
+          }
+        }
+        resolve(stdOut);
+      }
+    });
+  });
+
+  return compile.then(
+    async stdOut => {
+      const sourceMap: RawSourceMap = JSON.parse(await readFile(mapAsFile, 'utf8'));
+      return { code: stdOut, map: sourceMap };
+    },
+    (error: Error) => {
+      throw error;
+    },
+  );
+};
+
+export default function closureCompiler(outputOptions: OutputOptions = {}, compileOptions: CompileOptions = {}): Plugin {
```

Comment:
u might want to validate `outputOptions` - currently it has to be an object for a single `format`, but rollup accepts also arrays here - I guess this is a tricky part with providing some hook for output options, because **I guess** that u need them in `transform` hook, but I think rollup assumes that `transform` hook should return same code for each format (it shouldnt be concerned with final format)

Reply:
Good idea to check this. I went ahead and removed the double specification with a refactor of the `Transform` lifecycle.

This brings up a good point though, there are no tests for multiple output configuration.